.github/workflows/code-testing.yml

```
@@ -0,0 +1,82 @@
+---
+name: Linting and Testing Anta
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  check-requirements:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: install requirements
+        run: |
+          pip install .
+      - name: install dev requirements
+        run: pip install .[dev]
+  lint-yaml:
+    name: Run linting for yaml files
+    runs-on: ubuntu-20.04
+    steps:
```

Maybe Add constraint to run in correct order

```suggestion
    needs: [check-requirements]
    steps
```